### ISSUE TITLE
hicdiff error

### ISSUE THREAD

**Issue by lijiacd985** [dozmorovlab/HiCdiff#1]
I am encountering this error when I used hicdiff:

hic.table=hic_diff(hic.table,Plot=TRUE)
Error in numerator + 1 : non-numeric argument to binary operator

I checked the file format of hic.table , it is the same with your example.

Could you please give me some suggestions?

Thanks

**Comment by jstansfield0** [comment]
I would recommend you use the HiCcompare version of our package which is available here: https://github.com/dozmorovlab/HiCcompare as this is the most up to date version of the package.

As for you error, could you give me the results of head() on the hic.table object you are trying to perform hic_diff() on? Did you first normalize the data with hic_loess() before plugging it into it hic_diff()?

**Comment by lijiacd985** [comment]
Thanks for your reply.

I am using HiCcompare

For R codes:
> library(HiCcompare)
> a<-read.table("wt.100k")

> b<-read.table("tko.100k")

> chr2.table=create.hic.table(a,b,chr='chr2')

> hic.table=hic_loess(chr2.table,Plot=TRUE)

Span for loess: 0.215950929844704

GCV for loess: 4.25925805031969e-06

AIC for loess: 0.854572059899249

`geom_smooth()` using method = 'gam'

`geom_smooth()` using method = 'gam'

> head(hic.table)                                                                      

   chr1  start1    end1 chr2  start2    end2 IF1       IF2 D           M

1: chr2 3000000 3100000 chr2 3000000 3100000  89 105.08814 0  0.23972266

2: chr2 3000000 3100000 chr2 3100000 3200000 154 136.98990 1 -0.16886081

3: chr2 3100000 3200000 chr2 3100000 3200000 659 653.98603 0 -0.01101864

4: chr2 3000000 3100000 chr2 3200000 3300000  67  69.43324 2  0.05146534

5: chr2 3100000 3200000 chr2 3200000 3300000 294 279.60952 1 -0.07240266

6: chr2 3200000 3300000 chr2 3200000 3300000 415 384.69767 0 -0.10938626

     adj.IF1   adj.IF2       adj.M          mc

1:  88.25241 105.97835  0.26406212 -0.02433946

2: 153.05726 137.83368 -0.15114300 -0.01771781

3: 653.46444 659.52601  0.01332082 -0.02433946

4:  66.74040  69.70332  0.06266705 -0.01120171

5: 292.20022 281.33176 -0.05468484 -0.01771781

6: 411.51403 387.95648 -0.08504680 -0.02433946

> hic.table = hic_diff(hic.table, Plot = TRUE)

Error in numerator + 1 : non-numeric argument to binary operator

> head(hic.table)

   chr1  start1    end1 chr2  start2    end2 IF1       IF2 D           M

1: chr2 3000000 3100000 chr2 3000000 3100000  89 105.08814 0  0.23972266

2: chr2 3000000 3100000 chr2 3100000 3200000 154 136.98990 1 -0.16886081

3: chr2 3100000 3200000 chr2 3100000 3200000 659 653.98603 0 -0.01101864

4: chr2 3000000 3100000 chr2 3200000 3300000  67  69.43324 2  0.05146534

5: chr2 3100000 3200000 chr2 3200000 3300000 294 279.60952 1 -0.07240266

6: chr2 3200000 3300000 chr2 3200000 3300000 415 384.69767 0 -0.10938626

     adj.IF1   adj.IF2       adj.M          mc

1:  88.25241 105.97835  0.26406212 -0.02433946

2: 153.05726 137.83368 -0.15114300 -0.01771781

3: 653.46444 659.52601  0.01332082 -0.02433946

4:  66.74040  69.70332  0.06266705 -0.01120171

5: 292.20022 281.33176 -0.05468484 -0.01771781


Thanks

Best 

Jia Li

**Comment by lijiacd985** [comment]
Attached is the input data that I used. 

If possible, please try.


Thanks

Best

Jia Li

**Comment by jstansfield0** [comment]
Can you email me the data directly at stansfieldjc@vcu.edu?

It is not showing up in the github comment.

**Comment by jstansfield0** [comment]
I ran the following on the data you emailed me and did not get any errors. 

```
a <- read.table("tko.100k")
b <- read.table("wt.100k")
chr2.table=create.hic.table(a,b,chr='chr2')
hic.table=hic_loess(chr2.table,Plot=TRUE)
hic.table = hic_diff(hic.table, Plot = TRUE)
```

Are you sure you are using the most up to date version of the HiCcompare package? Try reinstalling the package from github and running your code again. 

I believe the error you are getting is due to how I originally iterated through distances and the error you got would occur if a distance did not have any non-zero IFs. This bug has been fixed in the latest version of the package.
